src/libs/test-libs/utillib/convert/testing.cpp

```diff
@@ -168,6 +168,18 @@ TEST_F(CThisTest, TestConverts_3) {
     return true;
 }}
 
+//------------------------------------------------------------------------
+TEST_F(CThisTest, TestConverts_4) {
+        cout << "Running " << testName << "\n";
+
+        ASSERT_EQ("double_2_Str1", double_2_Str(14.15019, 5), "14.15019")
+        ASSERT_EQ("double_2_Str2", double_2_Str(14.15019, 3), "14.150")
+        ASSERT_EQ("double_2_Str3", double_2_Str(10586816.10315913591, 10), "10586816.1031591352") // should be 10586816.1031591359???
+        ASSERT_EQ("double_2_Str4", double_2_Str(10586816.10315913591, 2), "10586816.10")
+        ASSERT_EQ("double_2_Str5", double_2_Str(1050, 3), "1050.000")
```

Comment:
Please add negative numbers and zero for both the float and the number of digits.

Reply:
Now that I think about it, this may be why I put that shirbird code in there to begin with. We need (and cannot do without) 18 decimal place accuracy. This is a hard stop.

In fact, the more I think about it, this is exactly why that original code was there to begin with. This, I think, is why Ethereum doesn't support floating point numbers.

I think this test is related https://github.com/Great-Hill-Corporation/quickBlocks/blob/287e7b3e624592c1a276044b65f7a92e849e08ff/src/libs/test-libs/etherlib/printFloat/printFloat.cpp#L32

Reply:
OK great. I will make sure we have accuracy. I am reading up on this stuff. Interestingly, both the old code and the new code fail the same tests. I'm digging into why.